core/crypto/src/signature.rs

```diff
@@ -556,6 +558,80 @@ impl<'de> serde::Deserialize<'de> for SecretKey {
 #[derive(Clone)]
 pub struct Secp256K1Signature([u8; 65]);
 
+impl Secp256K1Signature {
+    fn check_v(&self) -> Result<u8, crate::errors::ParseSignatureError> {
+        match self.0[64] {
+            27 | 28 => Ok(self.0[64] - 27),
+            _ => Err(crate::errors::ParseSignatureError::InvalidData {
+                error_message: "invalid `v` value".to_string(),
+            }),
+        }
+    }
```

Comment:
This interpretation of `v` is different from what the code below uses (specifically, `Signature::verify`). It is a very bad design if values the same type are interpreted differently by different functions.

Reply:
Ah, not sure why that hadn't clicked in my mind earlier that the 26 / 27 value is an ETH thing.

Reply:
This should be ok in this https://github.com/near/nearcore/pull/4380/commits/1f2a3141f83913957822ab45e4f421405519871f